studio/src/lib/schema-helpers.ts

```diff
@@ -650,9 +662,97 @@ export const getDeprecatedTypes = (
   return deprecatedTypes;
 };
 
+type ExtractedAuthenticationDirectives = {
+  authenticated: boolean;
+  requiresScopes: string[][] | null;
+}
+
+const extractAuthenticationDirectives = (
+    astNode: FieldDefinitionNode | TypeDefinitionNode | undefined | null
+) => {
+  const result: ExtractedAuthenticationDirectives = { authenticated: false, requiresScopes: null };
+  if (
+      !Array.isArray(astNode?.directives) ||
+      astNode.directives.length === 0
+  ) {
+    return result;
+  }
+
+  result.authenticated = astNode.directives.some((d) => d.name.value === "authenticated");
+
+  const requiresScopesDirectives = astNode.directives.filter((d) => d.name.value === "requiresScopes");
+  if (requiresScopesDirectives.length === 0) {
+    return result;
+  }
+
+  result.requiresScopes = requiresScopesDirectives
+      .filter(d => Array.isArray(d.arguments) && d.arguments.length > 0)
+      .flatMap(d =>
+          d.arguments!
+            .filter((arg: ConstArgumentNode) => arg.name.value === "scopes" && arg.value.kind === Kind.LIST)
+            .flatMap((arg: ConstArgumentNode) => (arg.value as ListValueNode).values)
+            .map((list: ListValueNode) => list.values.map((sv) => (sv as StringValueNode).value))
+      );
+
+  return result;
+}
+
+export const getAuthenticatedTypes = (
+    astSchema: GraphQLSchema,
+): GraphQLTypeDefinition[] => {
+  const authenticatedTypes: GraphQLTypeDefinition[] = [];
+
+  const checkType = (type: any) => {
+    const common = {
+      name: type.name,
+      description: type.description || "",
+      loc: type.astNode?.loc,
+    };
+
+    const authenticatedFields: ParsedGraphQLField[] = [];
+
+    let category: GraphQLTypeCategory | null = null;
+
+    if (
+        type instanceof GraphQLObjectType ||
```

Comment:
If type were not `any`, you could then match on `kind`, or am I mistaken?

Reply:
Sadly, `GraphQLInterfaceType` and `GraphQLObjectType` don't have the `kind` field as that is part of the `ast`. The common type, if we want to remove `any`, would be `GraphQLNamedType`

Reply:
My mistake. Follow up comment below incoming.